File: sources/fsharp/02_switch_case/02_switch_case.fs
// task 02 switch_case — expected output: 7500000075000000
// build: dotnet build -c Release 02_switch_case.fsproj    run: dotnet run -c Release (or bin/Release/net8.0/02_switch_case.dll)

let mutable acc = 0L

let mutable i = 0L
while i < 100000000L do
    match i % 4L with
    | 0L -> acc <- acc + 1L
    | 1L -> acc <- acc + i
    | 2L -> acc <- acc + 2L * i
    | _ -> acc <- acc + 3L * i
    i <- i + 1L

printfn "%d" acc

File: sources/csharp/02_switch_case/02_switch_case.cs
// task 02 switch_case — expected output: 7500000075000000
// build: coreclr: dotnet build -c Release | nativeaot: dotnet publish -c Release -p:PublishAot=true | mono: mcs -optimize+ 02_switch_case.cs    run: coreclr: dotnet run -c Release (or bin/Release/net8.0/02_switch_case.exe) | nativeaot: bin/Release/net8.0/publish/02_switch_case.exe | mono: mono 02_switch_case.exe
using System;

class Program
{
    static void Main()
    {
        long acc = 0;
        for (int i = 0; i < 100000000; i++)
        {
            switch (i % 4)
            {
                case 0:
                    acc += 1;
                    break;
                case 1:
                    acc += i;
                    break;
                case 2:
                    acc += 2 * i;
                    break;
                case 3:
                    acc += 3 * i;
                    break;
            }
        }
        Console.WriteLine(acc);
    }
}
